### DISCUSSION TITLE
Segmentation based on seed points

### DISCUSSION THREAD

**Discussion by jo-spek** [opengeos/segment-geospatial#305] [Q&A]
Hi everyone,

I've been trying extensively to find a solution to the following:

I would like to use existing point data as a basis for field delineation. Now I do not want all those points to be the foreground for one delineation, but every single point being one foreground for one delineation of it's own. Kind of like providing one foreground point in the input_prompts notebook, saving one segmentation, and continuing to the next one.

See in the following image what I mean. I provided multiple input points and the segmentation looks pretty decent...,
BUT: the five provided input points create one continuous polygon. The field boundaries are clearly visible in the background and I wanted 5 separate polygons matching those boundaries as output. That would actually work pretty well if I did one point at a time (I tried that successfully).
![image](https://github.com/user-attachments/assets/eda18bf4-4841-474c-9444-3ab13e27f860)

I tried with the Automatic Mask Generator, providing a custom grid instead of the default grid (e.g. 32x32 points per image). To my surprise though, the output on my custom grid delineated many field boundaries where my grid did not contain a point, and simultaneously did not delineate around some of the other points.
See what I mean here:
![image](https://github.com/user-attachments/assets/96cbd557-9b55-419b-a712-258189301760)

If anybody has a clue how to reach my goal to create multiple delineations each based on one seed, I'd be very grateful for your help!

See the Jupyter Notebook and my input points here: https://1drv.ms/f/s!As5ecqNfzS5GgtsRVRp4yvIqCC-1iw?e=c2fY8r

**Comment by giswqs** [comment]
Try using the `predict` function to do segmenation programmatically. 

https://samgeo.gishub.org/samgeo/#samgeo.samgeo.SamGeo.predict
